### ISSUE TITLE
After dragging the scrollbar, a selection is automatically performed when hovering the page

### ISSUE THREAD

**Issue by SoftVision-CosminMuntean** [mozilla-services/screenshots#2698]
[Affected versions]:
- Nightly 55.0a1 (Build ID:20170418030220)

[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Prerequisites]:
- Have latest Nightly build with Screenshots installed and a clean profile.
- The "extensions.screenshots.system-disabled" pref is set to false.

[Steps to reproduce]:
1. Open the browser and profile from prerequisites and navigate to any website (eg: imdb.com).
2. Click the Screenshots button from browser toolbar.
3. Click the Firefox scrollbar and scroll the page.
4. Release the mouse and hover the page content.
5. Observe the brower behavior.

[Expected result]:
- The Screenshots overlay is not dismissed and no selection is performed.

[Actual result]:
- Automatically a selection is performed when hovering the page content.

[Notes]:
- The issue is also reproducible on stage server.
- Here is a screen recording with the issue:
![selection vs scrollbar](https://cloud.githubusercontent.com/assets/19226534/25129849/7c5a82ce-2448-11e7-9fad-c9631a944d66.gif)

**Comment by ianb** [comment]
My googling about how to detect this case shows lots of hacky solutions and no good ones.

Maybe the best solution is just to ignore clicks that are near or past the edge of the screen.  Inboard scrollbars lead to clicks inside the page, on Windows the clicks will actually be outside the window.

Another option might be to specifically look for scroll events during mousedown/mouseup, and ignore the click in that case.  That seems more complicated, and might mess up the scroll behavior when you get near the edge of the screen.

**Comment by 6a68** [comment]
@ianb I wonder if this is a symptom of some bug with async scrolling? The latest comment in bug 1211610 indicates that APZ scrolling isn't yet ready to ride the trains. We might want to see if @SoftVision-CosminMuntean can repro on other channels before figuring out a workaround.

**Comment by ianb** [comment]
I remember this bug from a long time ago – it's the weird but apparently normal behavior that scrolling produces mousedown/mouseup events (maybe mousemove too?)

**Comment by SoftVision-CosminMuntean** [comment]
> We might want to see if @SoftVision-CosminMuntean can repro on other channels before figuring out a workaround.

This issue is reproducible on all channels (Firefox release, Beta, Aurora and Nightly).
I have tried to test also  older versions of Firefox, but the add-on is not working on versions < 51.0.

**Comment by dannycoates** [comment]
who even drags the scrollbar anymore besides QA folk? (my dad probably)

**Comment by ckprice** [comment]
Removing gamma blocker, Jared may check out later today.

**Comment by SoftVision-CosminMuntean** [comment]
I have retested this issue and is still reproducible on Windows and Linux using the latest Screenshots custom build on stage, created from Github repository.
But, it seems that the issue is no longer reproducible on Mac OS. Is there a chance that the fix is only applied on Mac OSs?

**Comment by ianb** [comment]
Yes, I thought the inboard scrollbars were only an issue on Macs, so the offset only applies to macs.  Probably this should be set to all platforms (but I don't have a system to confirm)

**Comment by chenba** [comment]
Fix for this is in Jared's comment over at https://github.com/mozilla-services/screenshots/pull/4401#issuecomment-387524590
